src/main/kotlin/de/cyface/collector/handler/MeasurementHandler.kt

```diff
@@ -55,194 +55,185 @@ import java.util.UUID
-                handleFirstChunkUpload(ctx, request, session, loggedInUser, contentRange, metaData, null)
-                return
-            }
-
-            // Search for previous upload chunk
-            val fs = ctx.vertx().fileSystem()
-            val path = session.get<Any>(UPLOAD_PATH_FIELD) as String
-            request.pause()
-            fs.exists(path).onSuccess { fileExists: Boolean? ->
-                try {
-                    request.resume()
-                    if (!fileExists!!) {
-                        session.remove<Any>(UPLOAD_PATH_FIELD) // was linked to non-existing file
-                        handleFirstChunkUpload(ctx, request, session, loggedInUser, contentRange, metaData, path)
-                        return@onSuccess
-                    }
-                    handleSubsequentChunkUpload(ctx, request, session, loggedInUser, contentRange, metaData, path)
-                } catch (e: RuntimeException) {
-                    ctx.fail(500, e)
-                }
-            }.onFailure { failure: Throwable? ->
-                LOGGER.error("Response: 500, failed to check if temp file exists")
-                ctx.fail(500, failure)
+            val checkResult = checkAndStore(session, loggedInUser, request.pipe(), contentRange, metaData)
+            checkResult.onSuccess { result ->
+                onCheckSuccessful(result, ctx, session)
             }
+            checkResult.onFailure(MeasurementFailureHandler(ctx))
         } catch (e: InvalidMetaData) {
-            LOGGER.error(String.format("Response: 422, %s", e.message), e)
+            LOGGER.error("Response: 422", e)
             ctx.fail(ENTITY_UNPARSABLE, e)
         } catch (e: Unparsable) {
-            LOGGER.error(String.format("Response: 422, %s", e.message), e)
+            LOGGER.error("Response: 422", e)
             ctx.fail(ENTITY_UNPARSABLE, e)
         } catch (e: IllegalSession) {
-            LOGGER.error(String.format("Response: 422, %s", e.message), e)
-            ctx.fail(ENTITY_UNPARSABLE, e)
-        } catch (e: PayloadTooLarge) {
-            LOGGER.error(String.format("Response: 422, %s", e.message), e)
+            LOGGER.error("Response: 422", e)
             ctx.fail(ENTITY_UNPARSABLE, e)
         } catch (e: SessionExpired) {
-            LOGGER.warn(String.format("Response: 404, %s", e.message), e)
+            LOGGER.warn("Response: 404", e)
             ctx.response().setStatusCode(NOT_FOUND).end() // client sends a new pre-request for this upload
         } catch (e: SkipUpload) {
             LOGGER.debug(e.message, e)
-            remove(session) // client won't resume
-            ctx.fail(PreRequestHandler.PRECONDITION_FAILED, e)
+            session.destroy() // client won't resume
+            ctx.fail(PRECONDITION_FAILED, e)
+        } catch (e: PayloadTooLarge) {
+            LOGGER.error("Response: 422", e)
+            // client won't resume
+            session.destroy()
```

Comment:
If it's possible to set the payload max size on runtime we could set the max size to something small and add a test which ensures that the correct response error is returned when trying to upload a file larger that this.

If not, I guess it's okay without, but we should at least test this manually once after pushing this on staging. You can create a task for me if you want.

Reply:
It seems to be wrong conceptually. The payload size is checked before an uploadIdentifier is loaded. This means either it is a new upload or a resume upload. If it is a new one we simply deny it and do not need to delete anything. If it is a resume the client was lying in its prior request, since a prior request was accepted. In this second case we should not delete since there seems to be some confusion on client side on how to resume. Maybe the initial request was correct and only the resume was erroneous. If the initial request was correct we would delete uploaded data even though it was a perfectly fine. I believe we should give the client the opportunity to send a correct request here. On a correctly implemented client, this should not occur and data garbage collection should clean up data uploaded by erroneous clients anyway.